### ISSUE TITLE
Incorrect definition of `float`

### ISSUE THREAD

**Issue by dpsanders** [dpsanders/ValidatedNumerics.jl#174]
`float(x::Interval) = convert(Interval{Float64}, x)` is wrong: when acting on `BigFloat` intervals, it should leave it as a `BigFloat`. 

In fact, since all `Interval`s now have `eltype`s that are `AbstractFloat`s, `float` should be a no-op.
(Since `float` is not for converting to `Float64`, but rather converting elements of an object to the nearest floating-point type.)

**Comment by lbenet** [comment]
Is your suggestion to get rid of the extension of `Base.float` to intervals (line 95 of `src/intervals/precision.jl`)? I think I agree. 

The only actual use of that method (that I have found) is in `src/intervals/conversion.jl`. Yet, it is within an extension of `convert`, so it may have further consequences...

**Comment by dpsanders** [comment]
Not to get rid of it, just to make it a no-op:

`float{T<:AbstractFloat}(x::Interval{T}) = x`

**Comment by dpsanders** [comment]
Related to this: should we actually restrict Intervals to having AbstractFloats inside now?

**Comment by dpsanders** [comment]
Currently I get

```
julia> Interval{Int}(3,3)
Error showing value of type ValidatedNumerics.Interval{Int64}:
------ StackOverflowError -------------- Stacktrace (most recent call last)

 [1] — round_string(::BigInt, ::Int64, ::RoundingMode{:Down}) at display.jl:64 (repeats 80000 times)

StackOverflowError:

julia> Interval{Rational{Int}}(3//4)
Segmentation fault: 11
```

which is pretty nasty!

**Comment by dpsanders** [comment]
Currently we can construct `Interval{Int}` but not do anything with them:

```
julia> a = Interval{Int}(3, 3);

julia> b = Interval{Int}(4,4);

julia> c = a + b;
MethodError: no method matching rounding_raw(::Type{Int64})
```

**Comment by dpsanders** [comment]
I think the correct definition, to take into account rational intervals, is

```
float(x::Interval) = (T = float(x.lo); convert(Interval{T}, x) )
```

**Comment by dpsanders** [comment]
Fixed in #188.
